fuzzbench/test_e2e_run.py

```diff
@@ -26,6 +28,15 @@
 class TestEndToEndRunResults:
     """Checks the result of a test experiment run."""
 
+    def test_jobs_dependency(self):
+        """Tests that jobs dependency preserves during working."""
+        jobs = {
+            name: Job.fetch(name, connection=redis.Redis(host='queue-server'))
+            for name in ['base-image', 'base-builder', 'base-runner']
+        }
+        assert jobs['base-image'].ended_at <= jobs['base-builder'].started_at
+        assert jobs['base-builder'].ended_at <= jobs['base-runner'].started_at
+
     def test_all_jobs_finished_sucessfully(self):
         """Fake test to be implemented later."""
```

Comment:
Consider implementing this check as well

Reply:
Good!